pkg/daemon/controller/controller.go

```diff
@@ -203,6 +203,130 @@ func (sc *ServiceController) Reconcile(_ context.Context, req ctrl.Request) (res
 	return ctrl.Result{}, nil
 }
 
+// Cleanup removes an instance from the cluster. If the upgradeInPlace flag is true, the node object is not deleted.
+// TODO: Remove/ensure files does not exist and also remove HNS networks, like WMCO does through windows.Deconfigure()
+func (sc *ServiceController) Cleanup(upgradeInPlace bool) error {
```

Comment:
I'd prefer unit tests that cover the functionality you are introducing be added within the same commit

Reply:
+1